### ISSUE TITLE
Add a tf.feature_column like sklearn.preprocessing.MultiLabelBinarizer

### ISSUE THREAD

**Issue by DachuanZhao** [tensorflow/tensorflow#42552]
**System information**
- TensorFlow version (you are using): 2.3
- Are you willing to contribute it (Yes/No):
If I can ......

**Describe the feature and the current behavior/state.**
```tf.feature_column``` doesn't have any column for dealing with multi-hot feature like ```sklearn.preprocessing.MultiLabelBinarizer``` now.

**Will this change the current api? How?**
I guess not

**Who will benefit with this feature?**
Everyone who has to deal with multi-hot feature 

**Any Other info.**

**Comment by DachuanZhao** [comment]
> @DachuanZhao Tensorflow does have an `indicator_column` which you can see [here](https://www.tensorflow.org/api_docs/python/tf/feature_column/indicator_column) that is used for multi-hot representation of a categorical column.
> Is that the feature you were looking for?

There are two questions : 
1.  I should give ```indicator_column``` the ```vocabulary_list```  explicitly .

```Python
# this is a df column
df_column_list = ["1,2","3"]
df_column_list = [v.split(",") for v in column_list]

# sklearn doesn't raise
from sklearn.preprocessing import MultiLabelBinarizer
mlb = MultiLabelBinarizer()
mlb.fit_transform(df_column_list)

#tf raise error 
test_column = tf.feature_column.categorical_column_with_vocabulary_list("category", df_column_list ,num_oov_buckets=0)
```

2. I should padding the ```df_column_list ``` first in tf .
```Python
test_column = tf.feature_column.categorical_column_with_vocabulary_list("category", ["1","2","3","4"],num_oov_buckets=0)

#This line raises error
#tf.keras.layers.DenseFeatures(tf.feature_column.indicator_column(test_column))({"category":[["1","2"],["2"],["3"],["4"],["3"]]})

#I should padding it first , then it doesn't raise error
tf.keras.layers.DenseFeatures(tf.feature_column.indicator_column(test_column))({"category":[["1","2"],["2","0"],["3","0"],["4","0"],["3","0"]]})
```
Both of them makes  ```indicator_column```  complicated to use ...

**Comment by DachuanZhao** [comment]
> @DachuanZhao I agree with @NikhilBartwal as he correct that it is being used for multi-hot represtation of the given categorical column!!
> I think it helped You!

> @DachuanZhao Tensorflow does have an `indicator_column` which you can see [here](https://www.tensorflow.org/api_docs/python/tf/feature_column/indicator_column) that is used for multi-hot representation of a categorical column.
> Is that the feature you were looking for?

There are two questions : 
1.  I should give ```indicator_column``` the ```vocabulary_list```  explicitly .

```Python
# this is a df column
df_column_list = ["1,2","3"]
df_column_list = [v.split(",") for v in column_list]

# sklearn doesn't raise
from sklearn.preprocessing import MultiLabelBinarizer
mlb = MultiLabelBinarizer()
mlb.fit_transform(df_column_list)

#tf raise error 
test_column = tf.feature_column.categorical_column_with_vocabulary_list("category", df_column_list ,num_oov_buckets=0)
```

2. I should padding the ```df_column_list ``` first in tf .
```Python
test_column = tf.feature_column.categorical_column_with_vocabulary_list("category", ["1","2","3","4"],num_oov_buckets=0)

#This line raises error
#tf.keras.layers.DenseFeatures(tf.feature_column.indicator_column(test_column))({"category":[["1","2"],["2"],["3"],["4"],["3"]]})

#I should padding it first , then it doesn't raise error
tf.keras.layers.DenseFeatures(tf.feature_column.indicator_column(test_column))({"category":[["1","2"],["2","0"],["3","0"],["4","0"],["3","0"]]})
```
Both of them makes  ```indicator_column```  complicated to use ...

**Comment by tanzhenyu** [comment]
We don't support feature column with Keras anymore (especially after 2.4) -- Please use `tf.keras.layers.experimental.preprocessing.CategoryEncoding` and its `adapt` method (which is equivalent to `fit_transform` here) and let us know if that fulfills your use case

**Comment by DachuanZhao** [comment]
> We don't support feature column with Keras anymore (especially after 2.4) -- Please use `tf.keras.layers.experimental.preprocessing.CategoryEncoding` and its `adapt` method (which is equivalent to `fit_transform` here) and let us know if that fulfills your use case

OK . It doesn't help me .
My problem is that I don't know how to deal with column like this :
```
import pandas as pd
df = pd.DataFrame( )
df["text_split_by_comma"] = ["1","2,3","4,5","1,6,7,9"]
```
There isn't any function for dealing with it in ```tf.data``` and ```tf.feature_column``` simply .

**Comment by DachuanZhao** [comment]
> We don't support feature column with Keras anymore (especially after 2.4) -- Please use `tf.keras.layers.experimental.preprocessing.CategoryEncoding` and its `adapt` method (which is equivalent to `fit_transform` here) and let us know if that fulfills your use case

Like this (https://github.com/tensorflow/tensorflow/issues/18572)[https://github.com/tensorflow/tensorflow/issues/18572]

**Comment by tanzhenyu** [comment]
You should read this as string and perform split *after* you read from csv, not during read from csv

**Comment by DachuanZhao** [comment]
> You should read this as string and perform split _after_ you read from csv, not during read from csv

It means I separate ```preprocess data```  from ```model``` if I do that . So I have to ```preprocess data```  once again when I call ```model.predict``` .

**Comment by tanzhenyu** [comment]
KPL can be part of your model:
```
class MyModel(tf.keras.Model):
  def __init__(self):
    self.preprocess_layer = ...
    ...
  def call(self, inputs):
    preprocessed_inputs = self.preprocess_layer(inputs)
    output = ...
    return output
```

**Comment by DachuanZhao** [comment]
> We don't support feature column with Keras anymore (especially after 2.4) -- Please use `tf.keras.layers.experimental.preprocessing.CategoryEncoding` and its `adapt` method (which is equivalent to `fit_transform` here) and let us know if that fulfills your use case

You should edit this [https://www.tensorflow.org/tutorials/structured_data/feature_columns](https://www.tensorflow.org/tutorials/structured_data/feature_columns) first if you don't  support feature column with Keras anymore ......

**Comment by tanzhenyu** [comment]
> > We don't support feature column with Keras anymore (especially after 2.4) -- Please use `tf.keras.layers.experimental.preprocessing.CategoryEncoding` and its `adapt` method (which is equivalent to `fit_transform` here) and let us know if that fulfills your use case
> 
> You should edit this https://www.tensorflow.org/tutorials/structured_data/feature_columns first if you don't support feature column with Keras anymore ......

That's not the way how this works. We need to move those experimental layers into core API before we can claim an alternative solution to be deprecated. For now, given they still serve a majority of use cases, they need to live.

But I do agree with the confusion, I think we can make this happen soon.

**Comment by DachuanZhao** [comment]
> > > We don't support feature column with Keras anymore (especially after 2.4) -- Please use `tf.keras.layers.experimental.preprocessing.CategoryEncoding` and its `adapt` method (which is equivalent to `fit_transform` here) and let us know if that fulfills your use case
> > 
> > 
> > You should edit this https://www.tensorflow.org/tutorials/structured_data/feature_columns first if you don't support feature column with Keras anymore ......
> 
> That's not the way how this works. We need to move those experimental layers into core API before we can claim an alternative solution to be deprecated. For now, given they still serve a majority of use cases, they need to live.
> 
> But I do agree with the confusion, I think we can make this happen soon.

You can refer to [https://scikit-learn.org/stable/modules/classes.html#module-sklearn.preprocessing](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.preprocessing) for ```tf.keras.layers.experimental.preprocessing```  , especially ```MultiLabelBinarizer``` and ```LabelBinarizer``` . It's really convenient~~~

**Comment by shangh1** [comment]
> KPL can be part of your model:
> 
> ```
> class MyModel(tf.keras.Model):
>   def __init__(self):
>     self.preprocess_layer = ...
>     ...
>   def call(self, inputs):
>     preprocessed_inputs = self.preprocess_layer(inputs)
>     output = ...
>     return output
> ```

I'm working on include my data processing code into the model, this is exactly what I'd like to use. But I just looked at functions in tf.keras.layers.experimental.preprocessing, it only has around 20 classes, seems very limited. My data processing code is written all in tf-functions, and I'm not able to convert all tf-functions in my data processing code to tf.keras.layers.experimental.preprocessing. Some of my tf functions in data processing code include: tf.strings.split, tf.lookup.StaticHashTable, tf.strings.to_number, tf.sparse.SparseTensor, tf.compat.v1.sparse_to_dense, tf.fill, tf.concat, tf.not_equal, tf.cast. Thanks.    @tanzhenyu @DachuanZhao

**Comment by DachuanZhao** [comment]
> > > We don't support feature column with Keras anymore (especially after 2.4) -- Please use `tf.keras.layers.experimental.preprocessing.CategoryEncoding` and its `adapt` method (which is equivalent to `fit_transform` here) and let us know if that fulfills your use case
> > 
> > 
> > You should edit this https://www.tensorflow.org/tutorials/structured_data/feature_columns first if you don't support feature column with Keras anymore ......
> 
> That's not the way how this works. We need to move those experimental layers into core API before we can claim an alternative solution to be deprecated. For now, given they still serve a majority of use cases, they need to live.
> 
> But I do agree with the confusion, I think we can make this happen soon.

Hi , I have read the newest document at [https://www.tensorflow.org/tutorials/structured_data/preprocessing_layers](https://www.tensorflow.org/tutorials/structured_data/preprocessing_layers) . It's very useful , but I have another question : 
How to serve the model in that document by tf-serving ? Is there a tf-serving document ready for this new feature ?
I add an issue at [https://github.com/tensorflow/tensorflow/issues/44002](https://github.com/tensorflow/tensorflow/issues/44002)  and you can answer it there .

**Comment by DachuanZhao** [comment]
> KPL can be part of your model:
> 
> ```
> class MyModel(tf.keras.Model):
>   def __init__(self):
>     self.preprocess_layer = ...
>     ...
>   def call(self, inputs):
>     preprocessed_inputs = self.preprocess_layer(inputs)
>     output = ...
>     return output
> ```

Hi , there are two issues about KPL :
[https://github.com/tensorflow/tensorflow/issues/44005](https://github.com/tensorflow/tensorflow/issues/44005)
[https://github.com/tensorflow/tensorboard/issues/4530](https://github.com/tensorflow/tensorboard/issues/4530)
Do you know anyone who may solve them ?  Could you please @  them in the issues ?
